rotkehlchen/db/upgrades/v40_v41.py

```diff
@@ -123,6 +125,43 @@ def _remove_covalent_api_key(write_cursor: 'DBCursor') -> None:
     log.debug('Exit _remove_covalent_api_key')
 
 
+def _move_labels_to_addressbook(write_cursor: 'DBCursor') -> None:
+    """Move all the `label` column values from `blockchain_accounts` table to the `name` column
+    of the 'address_book` table. If a `name` already exists in the `address_book` table, then
+    `address_name_priority` setting is used to determine which one to keep, defaults to
+    `DEFAULT_ADDRESS_NAME_PRIORITY`."""
+    log.debug('Enter _move_labels_to_addressbook')
+    address_name_priority = write_cursor.execute(  # get priority settings
+        'SELECT value FROM settings WHERE name = "address_name_priority"',
+    ).fetchone()
+    if address_name_priority is not None:
+        address_name_priority = json.loads(address_name_priority[0])
+    else:
+        address_name_priority = DEFAULT_ADDRESS_NAME_PRIORITY
+    address_book_priority = address_name_priority.index('private_addressbook')
+    blockchain_account_priority = address_name_priority.index('blockchain_account')
+    labels_to_move = {  # get all the labels from `blockchain_accounts` table
+        (blockchain, account): label for blockchain, account, label in write_cursor.execute(
+            'SELECT blockchain, account, label FROM blockchain_accounts WHERE label IS NOT NULL',
+        )
+    }
+    for address, blockchain, name in write_cursor.execute(
+        'SELECT address, blockchain, name FROM address_book',
+    ):  # replace with `address_book` names if its priority is higher
```

Comment:
here you are iterating all the address book that contains the velodrome, curve and many mroe entries. use a filter and query only for the addresses that you are moving, that is what you need right?